### DISCUSSION TITLE
How to autoResizePaper only on right and bottom side only?

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2183] [Q&A] [answered]
### Introduction

I've set `autoResizePaper: true` thus make if any object overlaps the paper, the paper will be resize (much larger)
My question is, is it possible `autoResizePaper` only on right side and bottom side of the paper?

![image](https://github.com/clientIO/joint/assets/106044262/516e5e7e-5d42-4764-b1ee-a8e78ffc5b7f)
Thank you

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
It can be done via the `contentOptions`.
```js
const scroller = new joint.ui.PaperScroller({
    paper,
    autoResizePaper: true,
    contentOptions: {
        allowNewOrigin: false
    }
});
```

**Reply by pandika-csgi** [reply]
I've used `JointJS v3.6.5 (2022-12-15)` and in the interface just can put these values `allowNewOrigin?: 'negative' | 'positive' | 'any';`
but yeah I use `positive` and it's worked
Is there any documentation for what the purpose of those values?

**Reply by kumilingus** [reply, marked as answer]
Does `null` work for you? You need to reset this to the default.
```ts
contentOptions: { allowNewOrigin: null }
```

See [docs](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#dia.Paper.prototype.getFitToContentArea).

**Reply by pandika-csgi** [reply]
Yeah `null` worked for me, Thanks @kumilingus

**Comment by pandika-csgi** [comment]
Hi @kumilingus , is there a way to make user cant drop an element to outside top and left side of paper?

**Reply by kumilingus** [reply]
Good question, you have to check it manually on `dragend' and cancel the drag if the position is wrong.

```js
stencil.on('element:dragend', function(elementView, evt, cloneArea) {
    if (cloneArea.x < 0 || cloneArea.y < 0) {
        stencil.cancelDrag({ dropAnimation: true });
    }
});
```
